packages/postcss-reduce-initial/src/script/lib/io.mjs

```diff
@@ -0,0 +1,37 @@
+import fetch from 'node-fetch';
+import { reduceInitial } from './mdnCssProps.mjs';
+
+export function validate(data) {
+  if (
+    data === undefined ||
+    !Object.keys(data.fromInitial).length ||
+    !Object.keys(data.toInitial).length
+  ) {
+    return Promise.reject(new Error('Fetching & processing JSON data failed!'));
+  }
+  return data;
+}
+
+export function handleError(error) {
+  if (error) {
+    throw error;
+  }
+}
+
+export function toJSONString(data) {
+  return `${JSON.stringify(data, null, 2)}\n`;
+}
+
+export function write(fileFunc, filePaths, data, key) {
+  fileFunc(filePaths[key], toJSONString(data[key]), handleError);
+  return data;
+}
+
+export async function generate(jsonURL, fileFunc, filePaths) {
```

Comment:
I feel injecting `fileFunc` makes the control flow harder to follow with no benefit since in practice `generate` only gets called once. If you want to make it testable in isolation I would  get rid of  `generate()` altogether, and call `fetch` and `writeFile` directly in `acquire.mjs`.

Reply:
Injecting the `fileFunc` allows the very basic function of `generate` ("should produce two files") to be asserted by a test, through passing in a spy instead of for example `fs.writeFile`. Without having to care about the specifics of the implementation. 

It also allows for the assertion mentioned in number 3 [here](https://github.com/cssnano/cssnano/pull/1160/files#r666879350) to be made.

Also commented more in full [here](https://github.com/cssnano/cssnano/pull/1160/files#r666914354).

I'd argue that not injecting, or moving these things to `acquire.mjs` would make proper test coverage more difficult.